src/forms/helpers/field-prop-types.js

```diff
@@ -17,10 +17,26 @@ export const fieldOptionsType = PropTypes.arrayOf(
         PropTypes.string,
         PropTypes.number
       ]).isRequired
-    })
+    }),
   ])
 )
 
+/**
+ *
+ * A constant representing the `PropTypes` of the `optionGroups` prop for select components, e.g., {@link Select} and {@link CheckboxGroup}
```

Comment:
Shouldn't mention `CheckboxGroup` here since it doesn't have the option group functionality (yet)